lib/host/wasi_nn/wasinnfunc.cpp

```diff
@@ -3,49 +3,478 @@
+          DimensionLen);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+
+    Status = ie_network_get_input_name(Network, Index, &InputName);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error(
+          "[WASI-NN] Unable to find input name correctly with Index:{}", Index);
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    /* Mark input as resizable by setting of a resize algorithm.
+     * In this case we will be able to set an input blob of any shape to an
+     * infer request. Resize and layout conversions are executed automatically
+     * during inference */
+    Status = ie_network_set_input_resize_algorithm(Network, InputName,
+                                                   RESIZE_BILINEAR);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to set input resize correctly");
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    Status = ie_network_set_input_layout(
+        Network, InputName,
+        layout_e::NHWC); // more layouts should be supported
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to set input layout correctly");
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+    Status = ie_network_set_input_precision(
+        Network, InputName,
+        precision_e::FP32); // more types should be supported
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to set input precision correctly");
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::InvalidArgument);
+    }
+
+    dimensions_t Dimens;
+    Dimens.ranks = DimensionLen;
+    for (size_t I = 0; I < Dimens.ranks; I++) {
+      Dimens.dims[I] = static_cast<size_t>(DimensionBuf[I]);
+    }
+
+    tensor_desc_t TensorDesc = {
+        layout_e::NHWC, Dimens,
+        precision_e::FP32}; // more types should be supported
+
+    Status = ie_blob_make_memory(&TensorDesc, &InputBlob);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to allocated input tensor correctly");
+      ie_blob_free(&InputBlob);
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::Busy);
+    }
+    Status = ie_blob_size(InputBlob, &BlobSize);
+    spdlog::debug("[WASI-NN] Blob size {}, with Tensor size {}", BlobSize,
+                  TensorDataLen / 4);
+    Status = ie_blob_get_cbuffer(InputBlob, &BlobCBuffer);
+    if (Status != IEStatusCode::OK) {
+      spdlog::error("[WASI-NN] Unable to find input tensor buffer");
+      ie_blob_free(&InputBlob);
+      ie_network_name_free(&InputName);
+      return static_cast<uint32_t>(WASINN::ErrNo::MissingMemory);
+    }
+
+    float *BlobData =
+        const_cast<float *>(static_cast<float const *>(BlobCBuffer.cbuffer));
```

Comment:
The `ie_blob_buffer` is a `union`:
https://docs.openvino.ai/cn/2021.1/ie_c_api/structie__blob__buffer.html

Why don't you use the `BlobCBuffer.buffer` directly instead of using `const_cast`?